loader.qml

```diff
@@ -1,109 +1,313 @@
 
     Rectangle {
-        id: myFrame
-        height: 640
-        width: frame.height
+        id: watchfaceDisplayFrame
+
         function snapshot() {
-            myFrame.grabToImage(
-            function(result) {  
-                result.saveToFile(testface + (round ? "-round.jpg" : ".jpg"))
-            }, Qt.size(320, 320) )
+            watchfaceDisplayFrame.grabToImage(function(result) {
+                result.saveToFile(approot.testface + (roundCheckBox.checked ? "-round.jpg" : ".jpg"));
+            }, Qt.size(320, 320));
         }
 
+        height: halfSize.checked ? 320 : 640
+        width: halfSize.checked ? 320 : 640
+
         Rectangle {
             id: frame
+
             anchors.fill: parent
             color: "black"
             focus: true
+            layer.enabled: roundCheckBox.checked
+            Keys.onReturnPressed: watchfaceDisplayFrame.snapshot()
+
             Image {
                 id: background
-                visible: !displayAmbient
+
+                visible: !approot.displayAmbient
                 source: "background.jpg"
                 anchors.fill: parent
             }
+
             Loader {
                 id: watchfaceLoader
+
                 anchors.fill: parent
-                source: testface + "/usr/share/asteroid-launcher/watchfaces/" + testface + ".qml"
+                source: approot.testface + "/usr/share/asteroid-launcher/watchfaces/" + approot.testface + ".qml"
             }
 
-            layer.enabled: round
-            layer.effect:  OpacityMask {
+            layer.effect: OpacityMask {
                 anchors.fill: parent
                 source: frame
+
                 maskSource: Rectangle {
                     width: frame.width
                     height: frame.height
-                    radius: frame.width/2
+                    radius: frame.width / 2
                 }
+
             }
 
-            Keys.onReturnPressed: snapshot()
         }
 
         Item {
             id: use12H
-            property var value: false
+
+            property var value: twelveHourCheckBox.checked
         }
 
         Item {
             id: wallClock
-            property var time: new Date()
+
+            property var time: getDisplayTime()
+
+            function getDisplayTime(statictime) {
+                var displayTime = new Date();
+                if (setStaticTimeCheckBox.checked) {
+                    displayTime.setHours(hoursTumbler.currentIndex, minutesTumbler.currentIndex, secondsTumbler.currentIndex);
+                    displayTime.setMonth(monthsTumbler.currentIndex, daysTumbler.currentIndex + 1);
+                }
+                return displayTime;
+            }
+
             Timer {
                 interval: 1000
                 running: true
                 repeat: true
-                onTriggered: wallClock.time = new Date()
+                onTriggered: wallClock.time = wallClock.getDisplayTime()
+            }
+
```

Comment:
I see you added a bunch of empty lines at the end of each block. Even though AsteroidOS does not really have a QML style guideline per-se, there is an official QML styleguide https://doc.qt.io/qt-5/qml-codingconventions.html and the "best practice" is to not add empty lines here. (same l95, l145 etc...)

I'm sorry that there are all these "unwritten rules" to obey :/ Hopefully that link should help you in the future serve as reference.

Reply:
I ran the files through [`qmlformat`](https://doc.qt.io/qt-6/qtquick-tools-and-utilities.html#qmlformat) version 1.0 exactly to avoid this kind of problem.  <sigh>